### ISSUE TITLE
problem with developer mode speed control [master]

### ISSUE THREAD

**Issue by GKart** [SuperTux/supertux#1914]
SuperTux 0.6.2-949
Windows 11 insider preview 22504 (although it was also tested on Windows 10 and arch linux with Linux zen 5.10)
Intel Celeron N4000, Intel UHD Graphics 600, 4GB DDR4 and 128GB Samsung SSD

By replicating the steps that will be given below, the game is expected to stop until it goes up to a speed greater than 0 (as if it happens in stable)

If you do it in a level the level is completed automatically but the creatures disappear, you regain absurd statistics and the statistics screen does not close automatically as it happens in stable but you must close it manually with the escape key
after that it will return to speed 100 [default speed] and the worldmap will behave correctly but when entering a level you will lose the ability to jump, even if you have fire x64 you will die from an attack, the animations will be bugged and if you kill to a creature the animation will be bugged

[this does not apply to the current stable version] activate the developer options and within one level access the debugging menu with the F2 key, then put a speed 0 [here you can also see that the speeds of the speed control are reversed] and exit that window, once that is done, the aforementioned problems will happen

**Comment by Semphriss** [comment]
Reproduced a segmentation fault. I presume the weird effects were the consequence of reading/writing data to the wrong memory; I would assume reproducing would give different effects each time, sometimes giving a crash.

Generally speaking, what is this feature used for? Given the difficulty there seems to be at implementing this properly, it might be worth considering removing it, if it does not serve a known purpose.

**Comment by GKart** [comment]
So far I tried it about 5 times and all with the same result
This allows you to skip scenes from the game but it does not fulfill a function that makes a big change so if it is eliminated it would not be a problem [in my opinion]

**Comment by GKart** [comment]
I just got the game to crash once the bug is applied go to the credits scene and lower the speed to 0%, once this is done go to the cheat menu and click on finish the level, once that is done the game will close without prior notice

**Comment by Rami-Slicer** [comment]
> Generally speaking, what is this feature used for? Given the difficulty there seems to be at implementing this properly, it might be worth considering removing it, if it does not serve a known purpose.

It might be useful for reproducing bugs that require precise timing but it's kind of finicky anyways.

**Comment by GKart** [comment]
> 

What seems strange to me is that in the current stable version it works perfectly, I don't know what changes they have made but before it was fine [or they changed something else and that affected him]

**Comment by Busybody64** [comment]
I already made an issue about 0% speed crashing the game:
https://github.com/SuperTux/supertux/issues/1867

**Comment by GKart** [comment]
I would never have found it if it was closed until 5 minutes ago xd

**Comment by Semphriss** [comment]
Closing as duplicate of #1867
